dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -177,13 +177,13 @@ public WorkflowDAO(SessionFactory factory) {
     }
 
     @SuppressWarnings({"checkstyle:ParameterNumber"})
-    public List<Workflow> filterTrsToolsGet(DescriptorLanguage descriptorLanguage, String registry, String organization, String name, String toolname,
+    public List<Workflow> filterTrsToolsGet(Class workflowType, DescriptorLanguage descriptorLanguage, String registry, String organization, String name, String toolname,
```

Comment:
Suggestion: `    public <T extends Workflow> List<T> filterTrsToolsGet(Class<T> workflowType, DescriptorLanguage descriptorLanguage, String registry, String organization, String name, String toolname,`. You'll have to make other changes too within, but I think they'll be clear.

Advantages:

1. The compiler enforces that you only call with a valid Class
2. The caller will get back a `List<Bioworkflow>` or `List<AppTool>` instead of a `List<Workflow>`. Which you don't care about today, but maybe someday.